credentials/service.js

```diff
@@ -35,6 +36,10 @@ const hooks = {
   }
 }
 
+function hasNoAgent (hook) {
+  return isNil(hook.data.agentId)
```

Comment:
in some cases it's okay to not pass in `agentId`, like an update to an existing agent. in those cases we need to lookup the existing object by id, get the agent id, then look up if the agent exists.

but wait, those cases probably shouldn't create an agent anyways, so yeah this is legit.